Code/Framework/AzToolsFramework/AzToolsFramework/PropertyTreeEditor/PropertyTreeEditor.cpp

```diff
@@ -601,7 +600,7 @@ namespace AzToolsFramework
                 if (keyAddress && valueAddress)
                 {
                     m_serializeContext->CloneObjectInplace(keyAddress, AZStd::any_cast<void>(&key), key.type());
-                    m_serializeContext->CloneObjectInplace(valueAddress, AZStd::any_cast<void>(&value), value.type());
+                    m_serializeContext->CloneObjectInplace(valueAddress, AZStd::any_cast<void>(&value), data.value().m_valueElement->m_typeId);
```

Comment:
We should verify that the source value is the type or derived from the destination type before copying.
We should be able to use the `data.value().m_valueElement.m_azRtti` to check if the type associated with the AZStd::any is derived from the the destination by using the `IRttiHelper::IsTypeOf` function

Reply:
Thanks, will do!

Reply:
Alright, I've been playing around with this but it turns out `value.type()` for strings will return the typeId `basic string)`, which isn't registered to the serialize context. So trying to use functions like `IsTypeOf`, or `m_serializeContext->FindClassData` etc. will not work with this. The conversion works, though.

So far, i've assumed that basic_string is intentionally not reflected, but I'm not sure if that's the case?

Reply:
@lumberyard-employee-dm is there an alternative to IsTypeOf that can let us identify conversions in this case?

Reply:
The basic_string type is registered with the SerializeContext implicitly when it is reflected as ->Field in the SerializeContext reflection.
So the typeid for is registered with the SerializeContext through the [SerializeGenericTypeInfo](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Serialization/AZStdContainers.inl#L3291-L3354) system